addons/im_livechat/static/src/new/chatbot/chatbot_service.js

```diff
@@ -0,0 +1,335 @@
+     * clear outdated storage.
+     */
+    async restore() {
+        const chatbotStorageKey = `im_livechat.chatbot.state.uuid_${this.livechatService.sessionCookie?.uuid}`;
+        const { _chatbotCurrentStep, _chatbot } = JSON.parse(
+            browser.localStorage.getItem(chatbotStorageKey) ?? "{}"
+        );
+        this.currentStep = _chatbotCurrentStep ? new ChatbotStep(_chatbotCurrentStep) : undefined;
+        this.chatbot = _chatbot ? new Chatbot(_chatbot) : undefined;
+        for (let i = 0; i < browser.localStorage.length; i++) {
+            const key = browser.localStorage.key(i);
+            if (key !== chatbotStorageKey && key.includes("im_livechat.chatbot.state.uuid_")) {
+                browser.localStorage.removeItem(key);
+            }
+        }
+    }
+
+    /**
+     * Save the chatbot state in the local storage.
+     */
+    async save() {
+        if (this.livechatService.state !== SESSION_STATE.PERSISTED) {
+            return;
+        }
+        browser.localStorage.setItem(
+            `im_livechat.chatbot.state.uuid_${this.thread.uuid}`,
+            JSON.stringify({
+                _chatbot: this.chatbot?.toServerData(),
+                _chatbotCurrentStep: this.currentStep?.toServerData(),
+            })
+        );
+    }
+
+    // =============================================================================
+    // GETTERS
+    // =============================================================================
+
+    get active() {
+        return this.available && this.thread?.isChatbotThread;
+    }
+
+    get available() {
+        return Boolean(this.chatbot);
+    }
+
+    get completed() {
+        return (
+            this.currentStep?.operatorFound ||
+            (this.currentStep?.isLast && !this.currentStep?.expectAnswer)
+        );
+    }
+
+    get inputEnabled() {
+        if (!this.active || this.currentStep?.operatorFound) {
+            return true;
+        }
+        return (
+            !this.isTyping &&
+            this.currentStep?.expectAnswer &&
+            this.currentStep?.answers.length === 0
+        );
+    }
+
+    get inputDisabledText() {
+        if (this.inputEnabled) {
+            return "";
+        }
+        if (this.completed) {
+            return _t("Conversation ended...");
+        }
+        switch (this.currentStep?.type) {
+            case "question_selection":
+                return _t("Select an option above");
+            default:
+                return _t("Say something");
+        }
+    }
+
+    get shouldRestore() {
+        // Use the uuid of the session cookie since the thread might not
+        // be created yet.
+        return Boolean(
+            localStorage.getItem(
+                `im_livechat.chatbot.state.uuid_${this.livechatService.sessionCookie?.uuid}`
+            )
+        );
+    }
+
+    get thread() {
+        return this.store.currentLivechatThread;
+    }
```

Comment:
It would be better to have explicit code that inserts the chatbot thread, so that this store entry is better named (could even not be in the store, it can be fully owned by chatbot service).

Reply:
The issue is that the `threadService` relies on the  `livechatService`: when calling `threadService.post`, we need to check whether the livechat session is persisted, persisting it otherwise. So, the `livechatService` can't rely on the `threadService` or it would lead to circular dependencies..